diff --git a/wrapping/src/CMakeLists.txt b/wrapping/src/CMakeLists.txt
index 94dab89a..d892c1b0 100644
--- a/wrapping/src/CMakeLists.txt
+++ b/wrapping/src/CMakeLists.txt
@@ -51,6 +51,12 @@ set(SWIG_MODULE_openmeeg_EXTRA_DEPS numpy.i make_geometry.py ${SWIG_MODULE_openm
 set_source_files_properties(openmeeg.i PROPERTIES CPLUSPLUS ON GENERATED_COMPILE_DEFINITIONS SWIG_PYTHON_SILENT_MEMLEAK)
 swig_add_library(openmeeg LANGUAGE python SOURCES openmeeg.i)
 
+# For PyPy, see
+# https://github.com/conda-forge/openmeeg-feedstock/pull/20#issuecomment-1161729565
+if(UNIX)
+    set_target_properties(openmeeg PROPERTIES SUFFIX ".${Python3_SOABI}.so")
+endif()
+
 if (APPLE)
     swig_link_libraries(openmeeg OpenMEEG::OpenMEEG)
 else()
