diff --git a/wrapping/python/CMakeLists.txt b/wrapping/python/CMakeLists.txt
index 4e89b3fb..a178065c 100644
--- a/wrapping/python/CMakeLists.txt
+++ b/wrapping/python/CMakeLists.txt
@@ -15,7 +15,8 @@ if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.27")
     cmake_policy(SET CMP0148 NEW)  # fail FindPythonInterp / FindPythonLibs
 endif()
 
-find_package(Python3 REQUIRED COMPONENTS Interpreter NumPy)
+find_package(Python3 REQUIRED COMPONENTS Interpreter NumPy Development.SABIModule)
+add_library (Python3::Module ALIAS Python3::SABIModule)
 
 if (BUILD_TESTING)
     find_package(Pytest)
